File: client/src/snow/Lifts.js
import React from 'react'

import { formatDateFull } from '../util/format.js'

export default function Lifts({
  lifts: { updatedOn, liftStatuses = [] } = {},
}) {
  return (
    <div className="lifts">
      <h2>
        Lifts <span className="updated">{formatDateFull(updatedOn)}</span>
      </h2>
      <ul className="lift-statuses">
        {liftStatuses.map((lift) => (
          <Lift lift={lift} key={lift.name} />
        ))}
      </ul>
    </div>
  )
}

function Lift({ lift: { name, status, hours } = {} }) {
  const hasSubstatus = name.includes('(')
  const isOpen = status && status.toLowerCase().includes('open')
  let subStatus
  if (hasSubstatus) {
    subStatus = name.substring(name.indexOf('('))
    name = name.substring(0, name.indexOf('(') - 1)
  }
  return (
    <li>
      <span className={`lift-name${isOpen ? ' open' : ''}`}>{name}</span>
      {subStatus && isOpen && (
        <span className="lift-substatus">{subStatus}</span>
      )}
      <span className="lift-status">{status}</span>
      {isOpen && <span className="lift-hours">{hours}</span>}
    </li>
  )
}

// {
//   "name": "BRUNO’S",
//   "status": "open",
//   "hours": "9:00am - 4:00pm"
// }

File: client/src/components/lifts.rs
use shared::Lifts;
use yew::prelude::*;

#[derive(Properties, PartialEq)]
pub struct Props {
    pub lifts: Lifts,
}

#[function_component(LiftsView)]
pub fn lifts_view(props: &Props) -> Html {
    let updated = props
        .lifts
        .updated_on
        .as_deref()
        .unwrap_or("Unavailable");

    html! {
        <div class="lifts">
            <h2>{"Lifts "}<span class="updated">{updated}</span></h2>
            <ul class="lift-statuses">
                { for props.lifts.lift_statuses.iter().map(|lift| {
                    let is_open = lift.status.to_lowercase().contains("open");
                    let (name, sub) = if lift.name.contains('(') {
                        let idx = lift.name.find('(').unwrap();
                        (lift.name[..idx].trim().to_string(), Some(lift.name[idx..].to_string()))
                    } else {
                        (lift.name.clone(), None)
                    };
                    let name_class = if is_open { "lift-name open" } else { "lift-name" };
                    html! {
                        <li key={lift.name.clone()}>
                            <span class={name_class}>{&name}</span>
                            if let Some(sub) = &sub {
                                if is_open {
                                    <span class="lift-substatus">{sub}</span>
                                }
                            }
                            <span class="lift-status">{&lift.status}</span>
                            if is_open {
                                <span class="lift-hours">{&lift.hours}</span>
                            }
                        </li>
                    }
                })}
            </ul>
        </div>
    }
}
